bluepysnap/circuit_ids.py

```diff
@@ -0,0 +1,222 @@
+
+        Args:
+            population (str): the population you want to extract.
+            inplace (bool): if set to True. Do the transformation inplace.
+
+        Returns:
+            CircuitNodeIds : a filtered CircuitNodeIds containing only IDs for the given population.
+        """
+        if not inplace:
+            return CircuitNodeIds(self.index[self._locate(population)], sort_index=False)
+        self.index = self.index[self._locate(population)]
+        return None
+
+    def get_populations(self, unique=False):
+        """Returns all population values from the circuit node IDs."""
+        if unique:
+            return self.index.levels[0].to_numpy()
+        return self.index.get_level_values(0).to_numpy()
+
+    def get_ids(self, unique=False):
+        """Returns all the ID values from the circuit node IDs."""
+        if unique:
+            return self.index.levels[1].to_numpy()
+        return self.index.get_level_values(1).to_numpy()
+
+    def copy(self):
+        """Copy a CircuitNodeIds."""
+        return CircuitNodeIds(self.index.copy())
+
+    def sort(self, inplace=False):
+        """Sort the CircuitNodeIds by population and then by ids.
+
+        Args:
+            inplace (bool): if set to True. Do the transformation inplace.
+
+        Notes:
+            sorting a CircuitNodeIds will gives better perf for the population extraction and the
+            data extraction from a sorted dataframe.
+        """
+        if not inplace:
+            return CircuitNodeIds(self.index, sort_index=True)
+        self.index = self.index.sortlevel()[0]
+        return None
+
+    def append(self, other, inplace=False):
+        """Append a CircuitNodeIds to the current one.
+
+        Args:
+            other (CircuitNodeIds): the other CircuitNodeIds to append to the current one.
+            inplace (bool): if set to True. Do the transformation inplace.
+        """
+        if not inplace:
+            return CircuitNodeIds(self.index.append(other.index))
+        self.index = self.index.append(other.index)
+        return None
+
+    def sample(self, sample_size, inplace=False):
+        """Sample a CircuitNodeIds.
+
+        Notes:
+            this function takes randomly ``sample_size`` values of a circuit node ids.
+
+        Args:
+            sample_size (int): the size of the sample. If the size of the sample is greater than
+                the size of the CircuitNodeIds then all ids are taken and shuffled.
+            inplace (bool): if set to True. Do the transformation inplace.
+        """
+        res = self if inplace else self.copy()
+        if len(res) != 0:
+            sample_size = sample_size if sample_size < len(res) else len(res)
+            res.index = res.index[np.random.choice(len(res), size=sample_size, replace=False)]
+        if not inplace:
+            return res
+        return None
+
+    def limit(self, limit_size, inplace=False):
```

Comment:
What about `resize` instead of `limit` ? Or `shrink`.
IMO what is missing in `limit` is the fact we are talking about size capacity. And not limit in the values themselves for example.

Reply:
I will stick to limit because it refers to the node API in NodePopulation: 
```
def ids(self, group=None, limit=None, sample=None)
```

If I change this I will change it for all the NodePopulation also.